### PR TITLE
Task/slug creation

### PR DESCRIPTION
**Pull request by jennywin**
@cchanningallen Parameterized slugs for everyone! Also fixed some authorization bugs. Will wait for your comments later and merge with master now to prep for today's v0 release.

### CODE REVIEW

File: app/controllers/tags_controller.rb [hunk 1]
```diff
@@ -1,14 +1,38 @@
...
+
+  def destroy
+    @tag.destroy
+  end
+
+  private
+    def set_tag
+      @tag = ActsAsTaggableOn::Tag.friendly.find(params[:id])
+      redirect_to action: 'show', id: @tag.friendly_id, status: 301 unless @tag.friendly_id == params[:id]
+    end
```
Comment (cchanningallen) [inline app/controllers/tags_controller.rb:33]: nice! DRY as :sunrise:

File: app/models/concerns/validate_tag.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module ValidateTag
```
Comment (cchanningallen) [inline app/models/concerns/validate_tag.rb:1]: Nice! Look at you, all fancy with `concerns`. :facepunch:

File: app/models/user.rb [hunk 3]
```diff
@@ -14,8 +14,8 @@ class User < ActiveRecord::Base
   
   validates :role, presence: true, inclusion: ROLES
 
-  def role?(role_query)
-    role == role_query
+  def role?(base_role)
+    role.present? && ROLES.index(base_role.to_s) <= ROLES.index(role)
```
Comment (cchanningallen) [inline app/models/user.rb:18]: what's going on here?

File: app/views/articles/_form.html.erb [hunk 4]
```diff
@@ -31,7 +31,7 @@
     </div>
 
     <div class="field">
-      <%= f.label :tag_list, "Tags" %>
+      <%= f.label :tag_list, 'Tags: allowed characters include [ a-z, 0-9, - ]' %>
```
Comment (cchanningallen) [inline app/views/articles/_form.html.erb:34]: It'd be rad to see these reused sections DRYed out into partials eventually

Comment (jennywin) [inline app/views/articles/_form.html.erb:34]: Oh yeah, I mentioned in a previous PR that I was going to break up a lot of stuff into partials as a separate task later on. I know it's super messy right now :/

**Comment by cchanningallen** [comment]
This looks _so cool_! Great work :)

Big +1

File: app/models/user.rb (hunk 3 above)
Comment (jennywin) [inline app/models/user.rb:18]: It was breaking something so I changed it back to the original code that was recommended by the CanCanCan gem for role based authorization.

**Pull request merged**